src/azure-cli/azure/cli/command_modules/vm/_validators.py

```diff
@@ -412,8 +412,12 @@ def _validate_vm_create_storage_profile(cmd, namespace, for_scale_set=False):
         else:
             raise CLIError('Unrecognized image type: {}'.format(image_type))
     else:
```

Comment:
Is it possible for a user to input both `--image` and `--attach-os-disk`? If so, please shrink the range of condition to not inputing neither `--image` nor `--attach-os-disk`.
```suggestion
    elif not namespace.image and not getattr(namespace, 'attach_os_disk', None):
```